api/src/models/JobMetadata.ts

```diff
@@ -58,31 +61,67 @@ class JobMetadata {
     }
 
     get derivativeStatus(): Record<string, unknown> {
-        // TODO: populate with real data
+        const lockfileExists: boolean = fs.existsSync(this.derivativeLockfile);
         const status = {
-            expected: 10,
+            expected: 0,
             processed: 0,
-            building: false,
+            building: lockfileExists,
         };
+        this.order.pages.forEach((page) => {
+            const image: Image = new Image(this.job.dir + "/" + page.filename);
+            Object.keys(image.sizes).forEach((key) => {
+                status.expected += 1;
+                if (fs.existsSync(image.derivativePath(key))) {
+                    status.processed += 1;
+                }
+            });
+        });
         return status;
     }
 
     get uploadTime(): number {
-        // TODO: populate with real data
-        return 0;
+        let mtime = 2000;
```

Comment:
It might be nice to use `const undefined_time = 2000;` above this, and then replace both hard-coded `2000`'s with `undefined_time`, just to eliminate a "magic number" from the code.